sdk/core/src/main/java/org/apache/mesos/offer/constrain/AgentRule.java

```diff
@@ -34,52 +40,162 @@ public String toString() {
         return String.format("AgentRule{agentId=%s}", agentId);
     }
 
+    @Override
+    public boolean equals(Object o) {
+        return EqualsBuilder.reflectionEquals(this, o);
```

Comment:
I would defer the change to the equals method until we encounter a situation where we need the SecurityManager enabled.